frontend/src/components/common/SideTab.tsx

```diff
@@ -1,10 +1,82 @@
-import React from "react";
-
-const SideTab = () => (
-  <>
-    SideTab
-    {/* TODO: SideTabを実装 */}
-  </>
-);
+import { AccountCircle, ListAlt, QrCode } from "@mui/icons-material";
+import {
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+} from "@mui/material";
+import { useRouter } from "next/router";
+
+const SideTab = () => {
+  const router = useRouter();
+  const handleClick = (link: string) => {
+    router.push(link).catch((error) => {
+      console.error(error);
+    });
+  };
+
+  const tabs = [
+    {
+      key: "home",
+      name: "ホーム",
+      icon: <AccountCircle />,
+      url: "/",
+    },
+    {
+      key: "personal-info",
+      name: "個人情報",
+      icon: <ListAlt />,
+      url: "/personal-info",
+    },
+    {
+      key: "qrcode",
+      name: "学祭QRコード",
+      icon: <QrCode />,
+      url: "/qrcode",
+    },
+  ];
```

Comment:
constantな要素はグローバルな領域に宣言した方が良いです
constantsディレクトリとかを作って、そこに定義するのもあり

```suggestion
  const TABS = [
    {
      key: "home",
      name: "ホーム",
      icon: <AccountCircle />,
      url: "/",
    },
    {
      key: "personal-info",
      name: "個人情報",
      icon: <ListAlt />,
      url: "/personal-info",
    },
    {
      key: "qrcode",
      name: "学祭QRコード",
      icon: <QrCode />,
      url: "/qrcode",
    },
  ];
const SideTab = () => {
  const router = useRouter();
  const handleClick = (link: string) => {
    router.push(link).catch((error) => {
      console.error(error);
    });
  };
```

Reply:
constants使うのはアリだな！
SideTabの内容をHead Tabでも使用しようとしてたのでちょうど良いかも
FinanSu参考にしてみます！